src/System.Net.Http/src/System/Net/Http/Managed/HttpConnectionPools.cs

```diff
@@ -34,8 +39,14 @@ internal sealed class HttpConnectionPools : IDisposable
 
         /// <summary>Initializes the pools.</summary>
         /// <param name="maxConnectionsPerServer">The maximum number of connections allowed per pool. <see cref="int.MaxValue"/> indicates unlimited.</param>
-        public HttpConnectionPools(int maxConnectionsPerServer)
+        
+        // CONSIDER: We are passing HttpConnectionSettings here, but all we really need are the SSL settings.
+        // When we refactor the SSL settings to use SslAuthenticationOptions, just pass that here.
```

Comment:
This will be nice, though we may want to just confirm for ourselves that we're ok with some of the implications.  In particular, today, for better or worse, there's a design on HttpClientHandler where once you make a request, "all" of the settings are fixed, and attempting to change a setting throws an exception.  I put "all" in quotes because I believe at least ClientCertificates semi-violates this: you're still not supposed to modify it once a request has been made, but I'm not aware of anything in the implementation that actually prevents you from doing so (though I expect due to the lack of synchronization that doing so could lead to corruption).  While I want to have the SslClientAuthenticationOptions and utilize that, doing so will lead to more properties being changeable after the first request, and if we don't copy all of those out into our own internal state bag, we could end up with options changing from request to request, more so than just ClientCertificates, but pretty much everything on SslClientAuthenticationOptions.  Something to think about, whether it matters, etc.

Reply:
Yep.  We could clone on first use -- that would freeze the settings (though it won't throw if you try to modify them).